### PR TITLE
add post replay with exp. regression, add settings, tests and require…

### LINKED ISSUE #4: Http post to post the replay results
it should implement exponential regression

### PR DESCRIPTION
**Pull request by bcattaneo**
Fixes #4

### CODE REVIEW

File: client.py [hunk 1]
```diff
@@ -1,23 +1,76 @@
...
+        body = {'title': replay_title, 'base64_file': base64_encoded_text, 'extra': extra_text}
+
+        # Send replay (HTTP post with exp. regression)
+        delay = 1
+        while (delay > 0):
+            try:
+                request = requests.post(self.replays_url, json=body)
+                delay = 0
+            except Exception as e:
+                sleep(math.exp(delay))
```
Comment (MMandirola) [inline client.py]: Generally we set a maximum number for delay, from that number on the delay doesn't increase, and the base of the regression generally is 2

File: test.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    # File to base64 string
+    base64_string = client.file_to_base64("/home/sc2/dev/Overlord/replay1.SC2Replay")
+
+    # Base64 string to file
+    print(client.base64_to_file(base64_string, "/home/sc2/dev/Overlord/replay1_copy.SC2Replay"))
+
+    # Send file
+    status_code = client.send_replay("overlord1.SC2Replay", base64_string, "")
+    print("sent" if (status_code == 201) else "not sent")
```
Comment (MMandirola) [inline test.py]: Unit test must have an assert

Comment (bcattaneo) [reply test.py]: For now its just a simple test file, not an unit test

**Comment by bcattaneo** [comment]
Also fixed #2

**Pull request merged**